internal/backend/local/test.go

```diff
@@ -1024,6 +1025,28 @@ func (runner *TestFileRunner) GetVariables(config *configs.Config, run *modulete
 		diags = diags.Append(valueDiags)
 	}
 
+	// We need to also process all global variables for tests
+	for name, value := range runner.Suite.GlobalTestVariables {
```

Comment:
The implementation here is fine, but we should only be applying the variables from the testing directory to tests also in the testing directory. 

You can access the tests filename from the `TestFile` argument, and you could modify the TestSuiteRunner struct so it also accepts the test directory. Then something like `if filepath.Dir(file.Name) == runner.Suite.TestDirectory { ... }` around the new for loop should work.